spot_ros2_control/hardware/include/spot_ros2_control/spot_hardware_interface.hpp

```diff
@@ -0,0 +1,131 @@
+#include "hardware_interface/hardware_info.hpp"
+#include "hardware_interface/system_interface.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "rclcpp/macros.hpp"
+#include "rclcpp_lifecycle/node_interfaces/lifecycle_node_interface.hpp"
+#include "rclcpp_lifecycle/state.hpp"
+#include "spot_ros2_control/visibility_control.h"
+
+#include "bosdyn/client/lease/lease_keepalive.h"
+#include "bosdyn/client/robot_command/robot_command_client.h"
+#include "bosdyn/client/robot_command/robot_command_helpers.h"
+#include "bosdyn/client/robot_command/robot_command_streaming_client.h"
+#include "bosdyn/client/robot_state/robot_state_client.h"
+#include "bosdyn/client/robot_state/robot_state_streaming_client.h"
+#include "bosdyn/client/sdk/client_sdk.h"
+#include "bosdyn/client/service_client/client_header_handling.h"
+#include "bosdyn/client/time_sync/time_sync_helpers.h"
+#include "bosdyn/client/util/cli_util.h"
+
+using StateHandler = std::function<void(::bosdyn::api::RobotStateStreamResponse&)>;
+
+namespace spot_ros2_control {
+
+class StateStreamingHandler {
+ public:
+  void handle_state_streaming(::bosdyn::api::RobotStateStreamResponse& robot_state);
+  std::vector<float> get_position();
+  std::vector<float> get_velocity();
+  std::vector<float> get_load();
+
+ private:
+  std::vector<float> current_position_;
+  std::vector<float> current_velocity_;
+  std::vector<float> current_load_;
+};
+class SpotHardware : public hardware_interface::SystemInterface {
+ public:
+  RCLCPP_SHARED_PTR_DEFINITIONS(SpotHardware)
+
+  SPOT_ROS2_CONTROL_PUBLIC
+  hardware_interface::CallbackReturn on_init(const hardware_interface::HardwareInfo& info) override;
+
+  SPOT_ROS2_CONTROL_PUBLIC
+  hardware_interface::CallbackReturn on_configure(const rclcpp_lifecycle::State& previous_state) override;
+
+  SPOT_ROS2_CONTROL_PUBLIC
+  std::vector<hardware_interface::StateInterface> export_state_interfaces() override;
+
+  SPOT_ROS2_CONTROL_PUBLIC
+  std::vector<hardware_interface::CommandInterface> export_command_interfaces() override;
+
+  SPOT_ROS2_CONTROL_PUBLIC
+  hardware_interface::CallbackReturn on_activate(const rclcpp_lifecycle::State& previous_state) override;
+
+  SPOT_ROS2_CONTROL_PUBLIC
+  hardware_interface::CallbackReturn on_deactivate(const rclcpp_lifecycle::State& previous_state) override;
+
+  SPOT_ROS2_CONTROL_PUBLIC
+  hardware_interface::CallbackReturn on_shutdown(const rclcpp_lifecycle::State& previous_state) override;
+
+  SPOT_ROS2_CONTROL_PUBLIC
+  hardware_interface::return_type read(const rclcpp::Time& time, const rclcpp::Duration& period) override;
+
+  SPOT_ROS2_CONTROL_PUBLIC
+  hardware_interface::return_type write(const rclcpp::Time& time, const rclcpp::Duration& period) override;
+
+ private:
+  // TODO(khughes): Hard coding this for now, but there should be a cleaner way to do this.
+  // The 3 interfaces are position, velocity, and effort.
+  size_t interfaces_per_joint_ = 3;
```

Comment:
Does this change? If not

```suggestion
  constexpr size_t interfaces_per_joint_ = 3;
```